slang-kotlin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -175,13 +173,8 @@ private Tree createElement(@Nullable PsiElement element) {
       }
       return new MatchCaseTreeImpl(getTreeMetaData(whenElement), conditions, body);
     } else if (element instanceof KtStringTemplateExpression) {
-      KtStringTemplateEntry[] entries = ((KtStringTemplateExpression) element).getEntries();
-      if (entries.length == 1 && entries[0] instanceof KtLiteralStringTemplateEntry) {
-        // Non-template strings, ie. not in the form "string ${1 + 1}"
-        return new LiteralTreeImpl(metaData, '\"' + entries[0].getText() + '\"');
-      } else {
-        return new NativeTreeImpl(metaData, new KotlinNativeKind(element), list(Arrays.stream(element.getChildren())));
-      }
+      // Do not differentiate between template and non-template strings for now
+      return new LiteralTreeImpl(metaData, element.getText());
```

Comment:
I believe that we lose some information when doing that. Ideally, when a template refers to an identifier, we should have this identifier somewhere in the AST (and it would be good to have a unit test for that). That's not a big issue for the rules which we have for the moment, so we could decide to leave that aside.